### PR TITLE
Project structure cleanup

### PR DESCRIPTION
**Pull request by sgdc3**
General maven project and modules cleanup.
Some minor enhancements to the CI configs.

TODO:
Fix one class shading conflict about netty (ETA: tomorrow) [Travis log](https://travis-ci.org/DragonetMC/DragonProxy/jobs/329277674#L2229)

### CODE REVIEW

**Comment by HoverEpic** [comment]
Wow, moved all files. Is it realy usefull to place all modules in a folder called modules ? I understand pom and CI update, but relocaling all files is realy needed ?

**Comment by DefinitlyEvil** [comment]
lel

**Comment by sgdc3** [comment]
mixing the screenshoot/doc folders with modules isn't very organized

**Comment by lukeeey** [comment]
I would personally keep it the same. We can remove the docs folder too

**Comment by ApocalypsjeNL** [comment]
Now I'm wondering, why do you recommend splitting the plugins? We can also create an universal plugin who can be used on BungeeCord and Spigot

**Comment by sgdc3** [comment]
that's why i created an universal subproject to create a superjar

**Comment by ApocalypsjeNL** [comment]
but it's already configured for a superjar?

File: api/pom.xml [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+
     <artifactId>dragonproxy-api</artifactId>
-    <version>0.0</version>
-    <name>DragonProxy-API</name>
-    <groupId>org.dragonet</groupId>
-    
-    <properties>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (ghost) [inline api/pom.xml:11]: These properties are removed, but never replaced. They're especially important for synchronization between OSes (Mac, Linux, Windows), and for Java bytecode versioning.

Comment (sgdc3) [reply api/pom.xml:11]: Encoding: https://github.com/sgdc3/DragonProxy/blob/master/pom.xml#L25
Compiler source version: https://github.com/sgdc3/DragonProxy/blob/master/pom.xml#L44

Comment (ghost) [reply api/pom.xml:11]: My bad, didn't see these

**Comment by sgdc3** [comment]
Any news? @HoverEpic

**Review by HoverEpic — APPROVED** [review]

**Comment by ApocalypsjeNL** [comment]
My question still applies, why did you split the plugin in different parts. currently it's configured as one universal plugin which can be installed on bungee and spigot at the same time. Why would you split them into different plugins AND a universal sone?

**Comment by HoverEpic** [comment]
It's for dependencies management, let bungeecord and spigot not interfer each other, some classes are the same, but in different package, it's more clear and prevnts import errors. (There are probably a more complex story about this, but this simple thing is ok for me to validate this)

**Review by DefinitlyEvil — APPROVED** [review]

**Comment by DefinitlyEvil** [comment]
:D

**Pull request merged**